frontend/test/components/AccountView/CredentialsBox.spec.js

```diff
@@ -0,0 +1,66 @@
+import {expect, describe, it, beforeEach, afterEach} from 'vitest';
+import {shallowMount} from '@vue/test-utils';
+import CredentialsBox from '@/components/AccountView/CredentialsBox.vue';
+import sinon from 'sinon';
+
+/**
+ * helper function that shallow mounts and returns the rendered component
+ * @param {props} props props for testing to overwrite default props
+ * @return {VueWrapper} returns a shallow mounted using props
+ */
+function getMountedComponent(props) {
+  const defaultProps = {
+    header: '',
+    clientId: '',
+    clientSecret: '',
+    content: [],
+  };
+
+  return shallowMount(CredentialsBox, {
+    props: {...defaultProps, ...props},
+  });
+}
+
+describe('CredentialsBox.vue', () => {
+  let displaySecretSpy;
+  let generateSecretSpy;
+
+  beforeEach(() => {
+    displaySecretSpy = sinon.spy(CredentialsBox.methods, 'displaySecret');
+    generateSecretSpy = sinon.spy(CredentialsBox.methods, 'generateSecret');
```

Comment:
Instead of testing the internals of the component to see if the methods are called, VueJS Test utills tracks events that components emit. https://test-utils.vuejs.org/guide/essentials/event-handling.html#asserting-complex-events  You can find other places in our code base as a reference for testing events.